### PR TITLE
RC3: Export placeholder types and type-related functions

### PR DESCRIPTION
**Pull request by tylerbutler**
This is a collection of changes to enable existing codebases using RC3 to more easily migrate to RC4. The following RC4+ changes are represented in this PR:

- https://github.com/microsoft/FluidFramework/pull/21003
- https://github.com/microsoft/FluidFramework/pull/20123
- https://github.com/microsoft/FluidFramework/pull/20859

### CODE REVIEW

File: packages/dds/sequence/src/index.ts [hunk 1]
```diff
@@ -81,6 +81,7 @@ export {
 	getTextAndMarkers,
 	ISharedString,
 	SharedString,
+	SharedString as SharedStringClass,
```
Comment (jason-ha) [inline packages/dds/sequence/src/index.ts:84]: This looks like a breaking change in a patch. :/

**Review by jason-ha — COMMENTED** [review]
I don't see anything in here that can't be done on the Loop side in little helper utility.
When it comes time for the RC4 update, in addition to the version changes the helper can be changed to re-export instead of providing these implementations.

File: packages/dds/sequence/src/index.ts (hunk 1 above)
Comment (tylerbutler) [reply packages/dds/sequence/src/index.ts:84]: My intent was to re-export the same class with a different name. Is that breaking? I guess the other question is if it's actually safe. :)

Comment (jason-ha) [reply packages/dds/sequence/src/index.ts:84]: That is breaking. Anyone using the old name won't find it. Or if you export something different, then they are getting something different.

**Comment by tylerbutler** [comment]
These changes are being handled on the consumer side, so closing this PR.

**Pull request closed without merging**